tests/parser/features/test_internal_call.py

```diff
@@ -362,3 +365,59 @@ def foo() -> int128:
     return self.bar(1)
 """
     assert_tx_failed(lambda: get_contract_with_gas_estimation(code), StructureException)
+
+
+def test_selfcall_with_value_private(get_contract_with_gas_estimation, assert_tx_failed):
```

Comment:
Let's change all these cases to use `pytest.mark.parametrize`.  You can comment each example with the specific functionality it's testing.  Here's an example: https://github.com/ethereum/vyper/blob/76afcb1f599704b748b16272924149a472623a74/tests/signatures/test_method_id_conflicts.py#L10-L60